code/datums/components/storage/concrete/extract_inventory.dm

```diff
@@ -0,0 +1,34 @@
+/datum/component/storage/concrete/extract_inventory
+	max_combined_w_class = WEIGHT_CLASS_TINY * 3
+	max_items = 3
+	insert_preposition = "in"
+	attack_hand_interact = FALSE
+	quickdraw = FALSE
+
+//These need to be false in order for the extract's food to be unextractable
+//from the inventory
+
+/datum/component/storage/concrete/extract_inventory/Initialize()
+	. = ..()
+	RegisterSignal(parent, COMSIG_TRY_STORAGE_CONSUME_CONTENTS, .proc/processCubes)
+	set_holdable(/obj/item/food/monkeycube)
+
+
+/datum/component/storage/concrete/extract_inventory/proc/processCubes(var/obj/item/slimecross/reproductive/P)
```

Comment:
```suggestion
/datum/component/storage/concrete/extract_inventory/proc/processCubes(obj/item/slimecross/reproductive/P)
```
this should work, cant be a var defined here, thats why we define an internal var. we just want to know what specifically we are looking for